proposals/low-level-struct-improvements.md

```diff
@@ -805,25 +730,119 @@ For example the last line of calculating *safe-to-escape* of returns will change
 
 > the safe-to-escape of all argument expressions including the receiver. **This will exclude all arguments that line up with parameters marked as [DoesNotEscape]**
 
-Misc Notes:
-- The  `DoesNotEscapeAttribute` will be defined in the `System.Runtime.CompilerServices` namespace.
-- The `DoesNotEscapeAttribute` cannot be combined with the `[ThisRefEscapes]` attribute, doing so results in an error.
-- The `DoesNotEscapeAttribute` will be emitted as a `modreq`
+Detailed Notes:
+- The `[DoesNotEscape]` and `[RefThisEscapes]` cannot be combined on the same method 
```

Comment:
Isn't `DoesNotEscape` the default for member functions of a `ref struct`? Trying to confirm my understanding here. If so, can we should have an example where one would want to have `DoesNotEscape` applied to a member function.

Reply:
For a instance member of a `struct` of any kind the `this` parameter is *safe-to-escape* to the calling method but *ref-safe-to-escape* within the current method. The `[DoesNotEscape]` annotation moves both to current method. 

An example where you would need it is the following: 

```c# 
ref struct MyType { 
  Span<int> _span;
  public MyType(Span<int> span) { }
  public void Method(ref ValueStringBuilder builder) { ... }
}
```

This method falls squarely into the [method arguments must match](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.2/span-safety.md#method-arguments-must-match) problem because both `this` and `builder` are *safe-to-escape* to the calling method. That means from the point of view of the compiler `builder` could end up smuggling out `ref` state from `this`. Yes realistically this can't happen do to no common data but compiler doesn't do field level analysis, it's just looking at the escape scopes of the locations.

This mean though practically that you run into problems like this: 

``` 
void Example(ref ValueStringBuilder builder) {
  var myType = new MyType(stackalloc int[42]); 
  
 // Error: builder and myType have different lifetimes
  myType.Method(ref builder); 
``` 

This can be fixed though by noting that `this` does not escape back into the parameters 

``` 
[DoesNotEscape]
public void Method(ref ValueStringBuilder builder) { ... }
```